applications/FluidDynamicsApplication/symbolic_generation/navier_stokes/weakly_compressible_navier_stokes_cpp_template.cpp

```diff
@@ -174,15 +174,15 @@ void SymbolicNavierStokes<TElementData>::AddBoundaryTraction(
 }
 
 template <>
-void SymbolicNavierStokes< SymbolicNavierStokesData<2,3> >::ComputeGaussPointLHSContribution(
-    SymbolicNavierStokesData<2,3>& rData,
+void WeaklyCompressibleNavierStokes< WeaklyCompressibleNavierStokesData<2,3> >::ComputeGaussPointLHSContribution(
+    WeaklyCompressibleNavierStokesData<2,3>& rData,
     MatrixType& rLHS)
 {
-    const double rho = rData.Density;
+    const auto& rho = rData.Density;
```

Comment:
why the auto? and the ? i think that from an optimizer point of view previous version was better

Reply:
Because now the `DENSITY` is a vector (one value for node). Is is not a `double` anymore.

Reply:
If I'm not wrong it's an `array_1d<double,NumNodes>`

Reply:
Done, as all the arrays in here are bounded I avoided the use of `auto` in here. With this we ensure that there are no expression assignments as the one you noted for the `vconv` and help the optimizer.